src/Pages/Income.cshtml

```diff
@@ -83,17 +83,22 @@ else
         <input type="hidden" name="handler" value="Filter" />
         <label class="text-light col-md-4">Period:</label>
         <div class="row justify-content-center">
-            <input class="col-md-3 col-sm-5 col-5" for="startDate" asp-for="StartDate" type="date" value="@Model.StartDate.ToShortDateString()" />
+            <input class="col-md-3 col-sm-5 col-5" onchange="handleFilter(this)"  for="startDate" min="2016-01-01" max="@DateTime.Now.AddMonths(1).ToShortDateString()" asp-for="StartDate" type="date" value="@Model.StartDate.ToShortDateString()" />
             <label class="text-light ml-2 mr-2"> - </label>
-            <input class="col-md-3 col-sm-5 col-5" for="endDate" asp-for="EndDate" type="date" value="@Model.EndDate.ToShortDateString()" />
+            <input class="col-md-3 col-sm-5 col-5" onchange="handleFilter(this)"  for="endDate" min="2016-01-01" max="@DateTime.Now.AddMonths(1).ToShortDateString()" asp-for="EndDate" type="date" value="@Model.EndDate.ToShortDateString()" />
```

Comment:
Note for myself: since DateTime.Now is called several times and it is costly operation, property inside controller with DateTime.Today should be created.

Reply:
Maybe I'll do this in separate branch after merging this one, not to create more chaos 😆